### PR TITLE
[TEAM2-476][TEAM2-472] Display Crosschain Exact Matches and Add Bridge Asset Section

### PR DESCRIPTION
**Pull request by derHowie**
Figma link (if any):
https://www.figma.com/file/SFfYzRoHlapEecS0LTLwhQ/Swap-Aggregator-V2?node-id=898%3A56888
https://www.figma.com/file/SFfYzRoHlapEecS0LTLwhQ/Swap-Aggregator-V2?node-id=28%3A28566

## What changed (plus any additional context for devs)
The currency select modal will now display exact matches on other networks when no applicable results are returned for the current network selection. Additionally, we now display a 'bridge' section when a verified asset is selected as the input and the selected network's results contain the corresponding bridged version of the asset in the verified section.

## Screen recordings / screenshots
iOS PoW
https://recordit.co/rxuKudh6ov
https://recordit.co/fjxphs6IFH
Android PoW
http://recordit.co/mVuqNV9blJ

## What to test
Confirm that network's with verified assets in common display the bridge asset section when applicable.
Confirm that crosschain exact matches display when a token is not available on the currently selected network but is present on others.

## e2e
I did not add e2e for this change because crosschain swap search results vary based on liquidity at the time of the search request. I am open to solutions for this issue if we deem additional testing necessary.

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels? (`team1/team2`, `critical path`, `release`, `dev QA`)
- [x] Did you test both iOS and Android?
- [x] If your changes are visual, did you check both the light and dark themes?
- [ ] Added e2e tests? If not, please specify why
- [x] If you added new critical path files, did you update the CODEOWNERS file?
- [x] If no `dev QA` label, did you add the PR to the QA Queue?

### CODE REVIEW

File: src/hooks/useSwapCurrencyList.ts [hunk 1]
```diff
@@ -487,6 +589,52 @@ const useSwapCurrencyList = (
...
+      const currentNetworkChainId = ethereumUtils.getChainIdFromNetwork(
+        network as Network
+      );
+      if (currentNetworkChainId !== searchChainId) {
+        // including goerli in our networks type is causing this type issue
+        // @ts-ignore
+        const exactMatch = crosschainVerifiedAssets[network as Network].find(
+          (asset: RT) => {
+            const symbolMatch =
+              asset?.symbol.toLowerCase() === searchQuery.toLowerCase();
```
Comment (estebanmino) [inline src/hooks/useSwapCurrencyList.ts:611]: we could use isLowerCase match

File: src/helpers/tokenSectionTypes.ts [hunk 2]
```diff
@@ -1,4 +1,6 @@
 export default {
+  bridgeTokenSection: '􀊝 Bridge',
+  crosschainMatchSection: '􀤆 On other networks',
   favoriteTokenSection: '􀋃 Favorites',
   lowLiquidityTokenSection: '􀇿 Low Liquidity',
   unverifiedTokenSection: '􀇿 Unverified',
```
Comment (estebanmino) [inline src/helpers/tokenSectionTypes.ts:6]: not your pr but wondering if we should move it to languages

Comment (skylarbarrera) [reply src/helpers/tokenSectionTypes.ts:6]: ticketed: https://linear.app/rainbow/issue/TEAM2-525/i18n-token-section-types

**Review by skylarbarrera — APPROVED** [review]
there is a slight lag for the bridge section to show up but otherwise working great

https://cloud.skylarbarrera.com/Screen-Recording-2022-10-11-13-27-54.mp4

**Comment by derHowie** [comment]
> there is a slight lag for the bridge section to show up but otherwise working great
> 
> https://cloud.skylarbarrera.com/Screen-Recording-2022-10-11-13-27-54.mp4

Yeah 😢 If it's any consolation, it's not a delay in the bridge section showing up but rather a delay in the entire list when network is switched. I experimented with presenting the skeleton during this period, but with fast internet this ends up looking like a flicker more than a smooth skeleton state. Open to suggestions.

**Comment by estebanmino** [comment]
seems like related to what @skylarbarrera said but there's a lag when selecting a token as input and then going to other network https://www.loom.com/share/d88f26e9d97b4db3a3e4a8c2ca1d8fa9 at 4:05

choosing badger dao on arbitrum and then selecting optimism as network is showing badger dao as bridge option for a second

File: src/hooks/useSwapCurrencyList.ts [hunk 3]
```diff
@@ -381,12 +444,23 @@ const useSwapCurrencyList = (
...
+    searchChainId,
+    isCrosschainSearch,
+    inputCurrency?.type,
+  ]);
 
   const { colors } = useTheme();
 
   const currencyList = useMemo(() => {
     const list = [];
+    let bridgeAsset = isCrosschainSearch
```
Comment (estebanmino) [inline src/hooks/useSwapCurrencyList.ts:459]: maybe we need this one to be in sync with `searchChainId` so we don't have mismatches

Comment (estebanmino) [reply src/hooks/useSwapCurrencyList.ts:459]: no this is not happening, just checked

Comment (derHowie) [reply src/hooks/useSwapCurrencyList.ts:459]: Fix: https://github.com/rainbow-me/rainbow/pull/4274/commits/ec3d18865e698493656fb51e50ffe3260cf03ff7
https://recordit.co/olLhccighm

**Review by estebanmino — APPROVED** [review]
🪨

**Pull request merged**